src/prefect/server/schemas/actions.py

```diff
@@ -598,10 +598,10 @@ class ConcurrencyLimitV2Update(ActionBaseModel):
 
     active: Optional[bool] = Field(None)
     name: Optional[str] = Field(None)
-    limit: Optional[int] = Field(None)
-    active_slots: Optional[int] = Field(None)
-    denied_slots: Optional[int] = Field(None)
-    slot_decay_per_second: Optional[float] = Field(None)
+    limit: Optional[NonNegativeInteger] = Field(None)
+    active_slots: Optional[NonNegativeInteger] = Field(None)
+    denied_slots: Optional[NonNegativeInteger] = Field(None)
+    slot_decay_per_second: Optional[NonNegativeFloat] = Field(None)
```

Comment:
We need validation UI-side as well. While a warning pops up if you provide an invalid input, there's nothing to stop you from saving it. 
<details>
<summary>Images</summary>
<img src="https://github.com/PrefectHQ/prefect/assets/42048900/b605e702-cee0-4e8c-a0d2-dacff3c43cd9">
<img src="https://github.com/PrefectHQ/prefect/assets/42048900/3c2a9c0e-9b19-439c-927a-dc4125a491e6".
</details>